ui/src/pages/version/components/forms/validation/schema.js

```diff
@@ -19,9 +20,9 @@ const resourceRequestSchema = (maxAllowedReplica) => yup.object().shape({
   gpu_request: yup
     .string()
     .typeError("GPU value is required")
-    .when("gpu_resource_type", {
-      is: (v) => v !== undefined && v !== "None",
-      then: yup.string().required("GPU value is required"),
+    .when("gpu_name", {
+      is: (v) => v !== undefined,
+      then: (_) => yup.string().matches(gpuRequestRegex, { message:'Valid GPU value is required'}).required("GPU value is required"),
```

Comment:
This validation was never working in the first place since `None` is selected by default, and the way this was implemented allows the `None` string to go through as a valid value (when in reality it shouldn't). I've implemented a quick fix here to allow only integers but a better implementation would be to use a validation context containing the list of allowable GPU values (this comes from the environments context and can be filtered out after an environment is selected), and then check if the given value exists within that list. This however needs more time (it's not rocket science though) and is left out of this PR.